src/app/Mappings/components/MappingBuilder/MappingBuilder.tsx

```diff
@@ -30,6 +32,16 @@ export const MappingBuilder: React.FunctionComponent<IMappingBuilderProps> = ({
   builderItems,
   setBuilderItems,
 }: IMappingBuilderProps) => {
+  const messageSelectBoth = 'You must select a source and target before adding another mapping.';
+  const messageExhausted = `All sources ${mappingType.toLowerCase()} have been mapped.`;
```

Comment:
This creates the messages "all sources network have been mapped" and "all sources storage have been mapped". We need to instead use actual conditions for the two words, so it is phrased "all source networks" and "all source datastores".

Reply:
Done